utils/update_translation_stats.py

```diff
@@ -58,95 +57,77 @@ def generate_translation_stats(po_dir, output_file):
-            header = header + ' ' + line
-        header_entries = header.split(',')
-        column_counter = 0
-        total_column = 0
-        translated_column = 0
-        while column_counter < len(header_entries):
-            if header_entries[column_counter].strip() == 'Total Source Words':
-                total_column = column_counter
-            elif header_entries[column_counter].strip() == 'Translated Source Words':
-                translated_column = column_counter
-            column_counter = column_counter + 1
+        COLUMNS = {
+            'filename': 'Filename',
+            'total': 'Total Source Words',
+            'translated': 'Translated Source Words'
+        }
+        result = csv.DictReader(csv.StringIO(stats_output), dialect='unix', skipinitialspace=True)
+        missing_cols = set(COLUMNS.values()) - set(result.fieldnames)
+        if missing_cols:
+            sys.exit(
+                'Column(s) "{}" not found in output of pocount'.format('", "'.join(missing_cols)))
 
         # Now do the actual counting for the current textdomain
-        for line in result:
-            cells = line.split(',')
-            po_filename = cells[0]
+        for row in result:
+            po_filename = row[COLUMNS['filename']]
             if po_filename.endswith('.po'):
-                entry = TranslationStats()
                 locale = regex_po.match(po_filename).group(1)
-                if locale in locale_stats:
-                    entry = locale_stats[locale]
-                entry.total = entry.total + int(cells[total_column])
-                entry.translated = entry.translated + \
-                    int(cells[translated_column])
+                entry = locale_stats[locale]
+                entry.total += int(row[COLUMNS['total']])
+                entry.translated += int(row[COLUMNS['translated']])
                 if entry.translated > entry.total:
-                    print('Error! Translated ' + str(entry.translated) + ' (+' + (cells[
-                          translated_column]) + ') is bigger than the total of ' + str(entry.total) + '(' + (cells[total_column]) + ')\n' + line)
+                    print(('Error! Translated {entry.translated} ({c_translated}) is bigger than ' +
+                           'the total of {entry.total} ({c_total}) on line {line}\n').format(
+                        entry=entry, c_translated=row[COLUMNS['translated']],
+                        c_total=row[COLUMNS['total']], line=result.line_num))
                     sys.exit(1)
                 locale_stats[locale] = entry
 
     print('\n\nLocale\tTotal\tTranslated')
     print('------\t-----\t----------')
 
     # The total goes in a [global] section and is identical for all locales
-    result = '[global]\n'
-    result = '{}total={}\n\n'.format(
-        result, list(locale_stats.values())[0].total)
+    result = ('# This file is managed by utils/update_translations_stats.py\n\n'
+              '[global]')
```

Comment:
Caught by testing:

```suggestion
              '[global]\n')
```

Reply:
Thanks. Looks like I have forgotten to run after this change.